src/mots/directory.py

```diff
@@ -0,0 +1,216 @@
+            if len(modules) > 1:
+                self.index[path] = [m for m in modules if not m.exclude_module_paths]
+        self.index = dict(self.index)
+
+        # Load people directory
+        people = list(self.config_handle.config["people"])
+        bmo_data = _get_bmo_data(people) if query_bmo else {}
+        self.people = People(people, bmo_data)
+        if self.people.serialized != list(self.config_handle.config["people"]):
+            logger.debug("People directory modified, updating configuration...")
+            self.config_handle.config["people"] = self.people.serialized
+
+    def query(self, *paths: str) -> QueryResult:
+        """Query given paths and return a list of corresponding modules."""
+        result = {}
+        rejected = []
+        for path in paths:
+            if not (self.repo_path / path).exists():
+                logger.warning(f"Path {path} does not exist, skipping.")
+                rejected.append(path)
+                continue
+            result[path] = self.index.get(self.repo_path / path, list())
+        logger.debug(f"Query {paths} resolved to {result}.")
+
+        return QueryResult(result, rejected)
+
+
+class QueryResult:
+    """Helper class to simplify query result interpretation."""
+
+    paths: list = None
+    path_map: dict = None
+    rejected_paths: list = None
+    modules: list = None
+    owners: list = None
+    peers: list = None
+
+    data_keys = {
+        "paths",
+        "rejected_paths",
+        "modules",
+        "owners",
+        "peers",
+    }
+
+    def __init__(self, result, rejected):
+        data = {k: set() for k in self.data_keys}
+        self.path_map = result
+
+        for path in self.path_map:
+            data["paths"].add(path)
+            data["modules"].update(self.path_map[path])
+
+        for module in data["modules"]:
+            data["owners"].update([Person(**o) for o in module.owners])
+            data["peers"].update([Person(**p) for p in module.peers])
+
+        data["rejected_paths"].update(rejected)
+
+        for key in data:
+            setattr(self, key, list(data[key]))
+
+    def __add__(self, query_result):
+        """Merge the data from both QueryResult objects."""
+        path_map = self.path_map.copy()
+        path_map.update(query_result.path_map)
+        rejected_paths = set.intersection(
+            set(self.rejected_paths),
+            set(query_result.rejected_paths),
+        )
+        return QueryResult(path_map, rejected_paths)
+
+    def __radd__(self, query_result):
+        """Call self.__add__ since the order of addition does not matter."""
+        return self.__add__(query_result)
+
+
+@dataclass
+class Person:
+    """A class representing a person."""
+
+    bmo_id: int
+    name: str = ""
+    info: str = ""
+    nick: str = ""
+    bmo_data: InitVar[dict] = None
+
+    def __post_init__(self, bmo_data):
+        """Refresh BMO data from BMO API."""
+        if bmo_data:
```

Comment:
This `if` statement is a smell that we have two different use paths here:
1. Where we just init the dataclass the "normal" way: `Person(0, "name", "info", "nick")`
2. Where we're parsing raw-ish output Bugzilla: `Person(0, bmo_data)`.

I'd recommend removing `__post_init__()` and instead adding a static `parse()` method, [kinda like phabricator emails](https://github.com/mozilla-conduit/phabricator-emails/blob/master/phabricatoremails/render/events/phabricator.py#L39-L51).

This gives you two benefits:
1. You're no longer setting `bmo_id` twice
2. The API of this class is easier to use, because you use either the constructor _or_ the dedicated `Person.parse()` static function depending on the use case you're using.